mlflow/models/model.py

```diff
@@ -29,55 +30,140 @@
+                }
+        """
+        return self._flavors
 
-    #: A dictionary mapping the flavor name to how to serve the model as that flavor. For example:
-    #:
-    #: .. code-block:: python
-    #:
-    #:     {
-    #:         "python_function": {
-    #:             "model_path": "model.pkl",
-    #:             "loader_module": "mlflow.sklearn",
-    #:             "python_version": "3.8.10",
-    #:             "env": "conda.yaml",
-    #:         },
-    #:         "sklearn": {
-    #:             "pickled_model": "model.pkl",
-    #:             "sklearn_version": "0.24.1",
-    #:             "serialization_format": "cloudpickle",
-    #:         },
-    #:     }
-    flavors: Dict[str, Any]
+    @property
+    def model_uri(self):
+        """
+        The ``model_uri`` of the logged model in the format ``'runs:/<run_id>/<artifact_path>'``.
+        """
+        return self._model_uri
 
-    #: The ``model_uri`` of the logged model in the format ``'runs:/<run_id>/<artifact_path>'``.
-    model_uri: str
+    @property
+    def model_uuid(self):
+        """
+        The ``model_uuid`` of the logged model,
+        e.g., ``'39ca11813cfc46b09ab83972740b80ca'``.
+        """
+        return self._model_uuid
 
-    #: The ``model_uuid`` of the logged model, e.g., ``'39ca11813cfc46b09ab83972740b80ca'``.
-    model_uuid: str
+    @property
+    def run_id(self):
+        """
+        The ``run_id`` associated with the logged model,
+        e.g., ``'8ede7df408dd42ed9fc39019ef7df309'``
+        """
+        return self._run_id
 
-    #: The ``run_id`` associated with the logged model, e.g., ``'8ede7df408dd42ed9fc39019ef7df309'``
-    run_id: str
+    @property
+    def saved_input_example_info(self):
+        """
+        A dictionary that contains the metadata of the saved input example, e.g.,
+        ``{"artifact_path": "input_example.json", "type": "dataframe", "pandas_orient": "split"}``.
+        """
+        return self._saved_input_example_info
 
-    #: A dictionary that contains the metadata of the saved input example, e.g.,
-    #: ``{"artifact_path": "input_example.json", "type": "dataframe", "pandas_orient": "split"}``.
-    saved_input_example_info: Optional[Dict[str, Any]]
+    @property
+    def signature_dict(self):
+        """
+        A dictionary that describes the model input and output generated by
+        :py:meth:`ModelSignature.to_dict() <mlflow.models.ModelSignature.to_dict>`.
+        """
+        warnings.warn(
+            "Field signature_dict is deprecated since v1.28.1. Use signature instead.",
+            category=FutureWarning,
+            stacklevel=2,
+        )
+        return self._signature_dict
+
+    @signature_dict.setter
+    def signature_dict(self, value):
+        warnings.warn(
+            "Field signature_dict is deprecated since v1.28.1. Use signature instead.",
+            category=FutureWarning,
+            stacklevel=2,
+        )
+        self._signature_dict = value
```

Comment:
I don't think we need a setter here; can we remove it?

Reply:
Sure